modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/ResourceOpenAPIParser.java

```diff
@@ -180,25 +180,15 @@ public static String getMethodAnnotations(
 					sb.toString() + "})");
 		}
 
-		List<JavaMethodParameter> javaMethodParameters =
-			javaMethodSignature.getJavaMethodParameters();
-
 		StringBundler sb = new StringBundler("");
 
-		for (JavaMethodParameter javaMethodParameter : javaMethodParameters) {
-			String parameterName = javaMethodParameter.getParameterName();
-
-			if (parameterName.equals("pagination")) {
-				sb.append(_addParameter(_findParameter(operation, "page")));
-				sb.append(_addParameter(_findParameter(operation, "pageSize")));
-			}
-			else if (parameterName.equals("sorts")) {
-				sb.append(_addParameter(_findParameter(operation, "sort")));
-			}
-			else {
-				sb.append(
-					_addParameter(_findParameter(operation, parameterName)));
+		for (Parameter parameter : operation.getParameters()) {
+			if (StringUtil.contains(parameter.getIn(), "header")) {
```

Comment:
I think the `in` parameter could contains one String between a set of them, for example "header" or "path".

As one of them is header, maybe we should use `equals` instead of `contains`